Tasks/SonarQubePreBuild/SonarQubePreBuildImpl.ps1

```diff
@@ -172,3 +190,95 @@ function IsFilePathSpecified
                 [StringComparison]::OrdinalIgnoreCase)
 }
 
+
+function GetVersion($uri, $headers)
+{
+    $version = $null
+
+    Try
+    {
+        $jsonResp = Invoke-RestMethod -Uri $uri -Method Get -Headers $headers
+
+        if ($jsonResp)
+        {
+            $version = $jsonResp.SonarQube.Version
+        }
+
+    }
+    Catch [System.Net.WebException]
+    {
+        Write-Verbose "WebException while trying to invoke $url. Exception msg:$($_.Exception.Message)"
+    }
+
+    return $version
+}
+
+#
+# Helper that returns the version number of the SonarQube server
+#
+function GetSonarQubeServerVersion()
+{
+    param([String][ValidateNotNullOrEmpty()]$serverUrl,
+          [String][ValidateNotNullOrEmpty()]$userName,
+          [String][ValidateNotNullOrEmpty()]$password)
+
+    Write-Host "Fetching SonarQube server version.."
+
+    $httpHeaders = @{}
+    $serverUri = New-Object -TypeName System.Uri -ArgumentList $serverUrl
+    $serverApiUri = New-Object -TypeName System.Uri -ArgumentList ($serverUri, "/api/system/info")
+
+    $base64Auth = [System.Convert]::ToBase64String([Text.Encoding]::ASCII.GetBytes("$($userName):$($password)"))
```

Comment:
Any SQ user should be able to run the tasks and run the analysis. In fact, if you create a new user it will be added by default to the SQ group "Sonar Users" and implicitly to the "Anyone" group. By default, "Anyone" can perform analysis and provision projects. So this is adding quite a heavy restriction. It seems strange that reading the version requires admin privileges - the version is on the front page where "Anybody" can browse to.